Source/Evolve/WarpXEvolve.cpp

```diff
@@ -370,7 +370,9 @@ WarpX::Evolve (int numsteps)
 
         // End loop on time steps
     }
-    multi_diags->FilterComputePackFlushLastTimestep( istep[0] );
+    if (istep[0] >= max_step || cur_time >= stop_time) {
+        multi_diags->FilterComputePackFlushLastTimestep( istep[0] );
```

Comment:
Thanks for the PR! Perhaps add a comment here that this check is needed for the Python version, which allows the Evolve routine to be called multiple times.